src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicy.ts

```diff
@@ -419,6 +404,41 @@ export default class VideoPriorityBasedPolicy implements VideoDownlinkBandwidthP
     this.optimalReceiveStreams = chosenStreams.slice();
   }
 
+  private getNetworkEventStartTimestamp() {
+    const numberOfParticipants = this.subscribedReceiveSet.size();
+    const prevEstimated = this.prevDownlinkStats.bandwidthEstimateKbps;
```

Comment:
I don't think this creates the logic we want either since `prevDownlinkStats` and `downlinkStats` are continually updated. For example estimations [1000, 1200, 1200, 1200, ...] will lead to BWE never increases. [100, 10000, 10001, 10000 ...] will additionally be stuck at 100. I'm also confused with how if we see e.g. BWE going down, we are resetting the `decrease` timestamp, but setting the network event timestamp  to the increase timestamp.

I think there is a lot of state here and it can be pretty confusing. I also think this is becoming hard to test. I recommend making an `/* internal */` class which composes the logic we want, with its own unit tests, and clear comments.